# Análise Exploratória dos Dados

In [1]:
import pandas as pd
from pathlib import Path
import seaborn as sns
import matplotlib.pyplot as plt 
import numpy as np

In [2]:
caminho_arquivo = Path(r'C:\Users\alexa\OneDrive\Área de Trabalho\projeto_a3\dados_tratados\obesity_cleaned2.csv')

In [3]:
dados = pd.read_csv(caminho_arquivo)
del dados['Unnamed: 0']
dados


Country  Year     Obesity (%)         Sex Region
0      Afghanistan  2016   5.5 [3.4-8.1]  Both Sexes  Other
1      Afghanistan  2016   3.2 [1.3-6.4]        Male  Other
2      Afghanistan  2016  7.6 [4.3-12.4]      Female  Other
3      Afghanistan  2015   5.2 [3.3-7.7]  Both Sexes  Other
4      Afghanistan  2015   3.0 [1.3-6.0]        Male  Other
...            ...   ...             ...         ...    ...
24565     Zimbabwe  1976   0.5 [0.1-1.3]        Male  Other
24566     Zimbabwe  1976  7.0 [3.3-12.4]      Female  Other
24567     Zimbabwe  1975   3.7 [1.7-6.5]  Both Sexes  Other
24568     Zimbabwe  1975   0.4 [0.1-1.3]        Male  Other
24569     Zimbabwe  1975  6.7 [3.0-12.2]      Female  Other

[24570 rows x 5 columns]

### Os dados entre homens e mulheres são parecidos? Informe com números.

In [4]:
dados.dtypes

Country        object
Year            int64
Obesity (%)    object
Sex            object
Region         object
dtype: object

In [5]:
dados['Obesity_mean'] = dados['Obesity (%)'].str.split('[').str[0]

In [8]:
dados.dtypes

Country         object
Year             int64
Obesity (%)     object
Sex             object
Region          object
Obesity_mean    object
dtype: object

In [13]:
dados.Country

0        Afghanistan
1        Afghanistan
2        Afghanistan
3        Afghanistan
4        Afghanistan
            ...     
24565       Zimbabwe
24566       Zimbabwe
24567       Zimbabwe
24568       Zimbabwe
24569       Zimbabwe
Name: Country, Length: 24570, dtype: object

In [15]:
dados['Obesity_mean'] = pd.to_numeric(dados['Obesity_mean'], errors='coerce')
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24570 entries, 0 to 24569
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Country       24570 non-null  object 
 1   Year          24570 non-null  int64  
 2   Obesity (%)   24570 non-null  object 
 3   Sex           24570 non-null  object 
 4   Region        24570 non-null  object 
 5   Obesity_mean  24066 non-null  float64
dtypes: float64(1), int64(1), object(4)
memory usage: 1.1+ MB


In [17]:
dados.head()

Country  Year     Obesity (%)         Sex Region  Obesity_mean
0  Afghanistan  2016   5.5 [3.4-8.1]  Both Sexes  Other           5.5
1  Afghanistan  2016   3.2 [1.3-6.4]        Male  Other           3.2
2  Afghanistan  2016  7.6 [4.3-12.4]      Female  Other           7.6
3  Afghanistan  2015   5.2 [3.3-7.7]  Both Sexes  Other           5.2
4  Afghanistan  2015   3.0 [1.3-6.0]        Male  Other           3.0

In [28]:
media_homens = dados[dados['Sex'] == 'Male']
media_homens = media_homens['Obesity_mean'].mean()
media_homens

9.332984293193716

A média de obesidade mundial dos homens é de 9.33%

In [29]:
media_mulheres = dados[dados['Sex'] == 'Female']
media_mulheres = media_mulheres['Obesity_mean'].mean()
media_mulheres

15.533433059087509

A média de obesidade mundial das mulheres é de 15.53%

In [31]:
amplitude_homens_mulheres = media_mulheres - media_homens
amplitude_homens_mulheres

6.200448765893793

A diferença do percetual de gordura mundial entre homens e mulheres é de 6.20%, ou seja, as mulheres tem tendência de ter maior percentual de gordura do que os homens.

### Qual o percentual médio de obesidade por sexo na América do Norte no ano de 2010

In [44]:
america_do_norte_2010 = dados[(dados['Region'] == 'North America') & (dados['Year'] == 2010)]
america_do_norte_2010

Country  Year       Obesity (%)         Sex  \
3924                     Canada  2010  26.0 [23.2-28.9]  Both Sexes   
3925                     Canada  2010  25.8 [21.8-30.1]        Male   
3926                     Canada  2010  26.2 [22.4-30.2]      Female   
13878                    Mexico  2010  25.8 [23.1-28.6]  Both Sexes   
13879                    Mexico  2010  21.1 [17.5-25.0]        Male   
13880                    Mexico  2010  29.8 [25.8-33.9]      Female   
23454  United States of America  2010  32.3 [29.6-35.1]  Both Sexes   
23455  United States of America  2010  31.2 [27.4-35.3]        Male   
23456  United States of America  2010  33.3 [29.6-37.1]      Female   

              Region  Obesity_mean  
3924   North America          26.0  
3925   North America          25.8  
3926   North America          26.2  
13878  North America          25.8  
13879  North America          21.1  
13880  North America          29.8  
23454  North America          32.3  
23455  North America          31.2  
23456  North America          33.3

In [50]:
media_homens_na = america_do_norte_2010[america_do_norte_2010['Sex'] == 'Male']
media_homens_na.Obesity_mean.mean()

26.033333333333335

A média do percentual de pessoas do sexo masculino na América do norte no ano de 2010 é de 26.03%

In [51]:
media_mulheres_na = america_do_norte_2010[america_do_norte_2010['Sex'] == 'Female']
media_mulheres_na.Obesity_mean.mean()

29.766666666666666

A média do percentual de pessoas do sexo masculino na América do norte no ano de 2010 é de 29.77%

### Qual os top 3 com maior e menor taxa de aumento de índices de obesidade nesse período de 2010? E em 2016?

In [52]:
dados

Country  Year     Obesity (%)         Sex Region  Obesity_mean
0      Afghanistan  2016   5.5 [3.4-8.1]  Both Sexes  Other           5.5
1      Afghanistan  2016   3.2 [1.3-6.4]        Male  Other           3.2
2      Afghanistan  2016  7.6 [4.3-12.4]      Female  Other           7.6
3      Afghanistan  2015   5.2 [3.3-7.7]  Both Sexes  Other           5.2
4      Afghanistan  2015   3.0 [1.3-6.0]        Male  Other           3.0
...            ...   ...             ...         ...    ...           ...
24565     Zimbabwe  1976   0.5 [0.1-1.3]        Male  Other           0.5
24566     Zimbabwe  1976  7.0 [3.3-12.4]      Female  Other           7.0
24567     Zimbabwe  1975   3.7 [1.7-6.5]  Both Sexes  Other           3.7
24568     Zimbabwe  1975   0.4 [0.1-1.3]        Male  Other           0.4
24569     Zimbabwe  1975  6.7 [3.0-12.2]      Female  Other           6.7

[24570 rows x 6 columns]

In [57]:
dados_2010 = dados[dados['Year'] == 2010]
dados_2010[dados_2010['Obesity_mean'].sort_values(ascending=False)]

KeyError: "None of [Index([61.9, 58.9, 56.7, 56.1, 55.9, 54.8, 52.6, 52.5, 51.8, 51.7,\n       ...\n        nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan],\n      dtype='float64', length=585)] are in the [columns]"